applications/GeoMechanicsApplication/custom_elements/transient_Pw_line_element.h

```diff
@@ -356,48 +346,32 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) TransientPwLineElement : public Elem
-        array_1d<double, TDim> BodyAcceleration;
+        array_1d<double, TNumNodes * TDim> volume_acceleration;
+        GeoElementUtilities::GetNodalVariableVector<TDim, TNumNodes>(volume_acceleration, GetGeometry(), VOLUME_ACCELERATION);
+        array_1d<double, TDim> body_acceleration;
 
         array_1d<double, TNumNodes> fluid_body_vector = ZeroVector(TNumNodes);
         for (unsigned int integration_point_index = 0;
              integration_point_index < GetGeometry().IntegrationPointsNumber(GetIntegrationMethod());
              ++integration_point_index) {
             GeoElementUtilities::InterpolateVariableWithComponents<TDim, TNumNodes>(
-                BodyAcceleration, r_N_container, VolumeAcceleration, integration_point_index);
+                body_acceleration, rNContainer, volume_acceleration, integration_point_index);
 
-            array_1d<double, TDim> tangent_vector = CalculateTangentialElementUnitVectorOnIntegrationPoint(integration_point_index, J_container);
+            array_1d<double, TDim> tangent_vector = column(J_container[integration_point_index], 0);
             array_1d<double, 1> projected_gravity = ZeroVector(1);
-            projected_gravity(0) = MathUtils<double>::Dot(tangent_vector, BodyAcceleration);
-
-            const auto N = Vector{row(r_N_container, integration_point_index)};
-            const BoundedMatrix<double, TNumNodes, 1> Temp = prod(dN_dX_container[integration_point_index], constitutive_matrix);
+            projected_gravity(0) = MathUtils<double>::Dot(tangent_vector, body_acceleration);
+            const auto N = Vector{row(rNContainer, integration_point_index)};
             double RelativePermeability = mRetentionLawVector[integration_point_index]->CalculateRelativePermeability(RetentionParameters);
-            fluid_body_vector += r_properties[DENSITY_WATER] / r_properties[DYNAMIC_VISCOSITY] * RelativePermeability 
-                * prod(Temp, projected_gravity) * rIntegrationCoefficients[integration_point_index];
+            fluid_body_vector += r_properties[DENSITY_WATER] * RelativePermeability 
+                * prod(prod(rShapeFunctionGradients[integration_point_index], constitutive_matrix), projected_gravity) *
+                rIntegrationCoefficients[integration_point_index] / r_properties[DYNAMIC_VISCOSITY];
         }
         return fluid_body_vector;
     }
 
 
-    array_1d<double, TDim> CalculateTangentialElementUnitVectorOnIntegrationPoint(
-        const unsigned int integration_point_index,
-        const GeometryType::JacobiansType& r_J_container) const
-    {
-        array_1d<double, TDim> result;
-        if constexpr (TDim == 2) {
-            result = column(r_J_container[integration_point_index], 0);
-        } 
-        else if constexpr (TDim == 3) {
-            MathUtils<double>::CrossProduct(result, column(r_J_container[integration_point_index], 0), column(r_J_container[integration_point_index], 1));
-        }
-        return result;
-    }
```

Comment:
@rfaasse Thank you very much for the efforts reviewing this work.

In some cases, I started to find that some of 3D cases are not stable. After debugging, it becames clear to me that the Jacobian in 3D for line element includes only 1 column,  a matrix of 3x1. So, it was not correctly implemented for 3D.

The idea was taken from 3D element with surface condition, where the Jacobian rotates the 2D surface. Therefore, it includes 2 columns, with the size of 3x2. So, we need to have cross product of the first and second column. However, here, it is line element in 3D space. 

So, actually I believe it was implemented wrongly, and only the first column of Jacobian defines the rotation here. Then the case becomes same as 2D space, and for both :  

    result = column(r_J_container[integration_point_index], 0);

Then as it includes only one line and can be calculated directly, I found it unnecessary to defind an extra function. I included it in CalculateFluidBodyVector and removed that function.

Reply:
Okay clear, it's something to keep in mind if we use this element for 2D/3D as well instead of only for line elements. It could probably also be fixed by having CalculateTangentialElementUnitVectorOnIntegrationPoint check the local dimension instead of TDim and returning `column(r_J_container[integration_point_index], 0)` if local dimension is 1 or 2, and still do cross when local dimension is 3

FYI @avdg81 @WPK4FEM